packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -565,9 +576,88 @@ export function authorizationServiceBuilder(
         )
       );
     },
+
+    async createKeys({
+      clientId,
+      authData,
+      keysSeeds,
+      correlationId,
+      logger,
+    }: {
+      clientId: ClientId;
+      authData: AuthData;
+      keysSeeds: ApiKeysSeed;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<{ client: Client; showUsers: boolean }> {
+      logger.info(`Creating keys for client ${clientId}`);
+      const client = await retrieveClient(clientId, readModelService);
+      assertOrganizationIsClientConsumer(
+        unsafeBrandId(authData.organizationId),
+        client.data
+      );
+      assertKeyIsBelowThreshold(
+        clientId,
+        client.data.keys.length + keysSeeds.length
+      );
+      if (!client.data.users.includes(authData.userId)) {
+        throw userNotFound(authData.userId, authData.selfcareId);
+      }
+
+      await assertUserSelfcareSecurityPrivileges(
+        authData.selfcareId,
+        authData.userId,
+        authData.organizationId,
+        selfcareV2Client,
+        client.data.users.find((userId) => userId === authData.userId)
```

Comment:
What's the need for this? It seems you are looking for a `userId` which is equal to the one in `authData`. So, you could just pass `authData.userId`. There is already another check [above](https://github.com/pagopa/interop-be-monorepo/pull/575/files#diff-dbdfef4027ce7e4ab4a5697de6c66e5364588bf66e77ff0558e90d05e496dfd2R603-R605), to see whether this `userId` is included in the client.